src/main/default/lwc/lookup/lookup.js

```diff
@@ -63,6 +66,14 @@ export default class Lookup extends LightningElement {
             }
             return result;
         });
+
+        if (this.newSobjectName) {
```

Comment:
Rather than modifying the list results, let's add the option to create a new record in the HTML template directly. You can add that just after the iteration template.

This way we can have a dedicated click handler for record creation and simplify the code and tests.